e2e-polybft/e2e/nexus_bridge_test.go

```diff
@@ -145,3 +149,943 @@ func TestE2E_ApexBridge_Nexus(t *testing.T) {
+			newAmount, err = cardanofw.GetEthAmount(ctx, apex.Nexus, userNexus)
+			require.NoError(t, err)
+			fmt.Printf("Current Amount Nexus: %v. Expected: %v\n", newAmount, expectedAmount)
+
+			if newAmount.Cmp(expectedAmount) >= 0 {
+				break
+			}
+
+			time.Sleep(time.Second * 10)
+		}
+
+		fmt.Printf("Success count: %v. prevAmount: %v. newAmount: %v. expectedAmount: %v\n", succeededCount, ethBalanceBefore, newAmount, expectedAmount)
+	})
+
+	//nolint:dupl
+	t.Run("From Prime to Nexus 1000x 20min 90%", func(t *testing.T) {
+		sendAmount := uint64(1)
+		sendAmountDfm := new(big.Int).SetUint64(sendAmount)
+		exp := new(big.Int).Exp(big.NewInt(10), big.NewInt(6), nil)
+		sendAmountDfm.Mul(sendAmountDfm, exp)
+
+		sendAmountEth := ethgo.Ether(sendAmount)
+
+		ethBalanceBefore, err := cardanofw.GetEthAmount(ctx, apex.Nexus, userNexus)
+		fmt.Printf("ETH Amount before Tx %d\n", ethBalanceBefore)
+		require.NoError(t, err)
+
+		instances := 1000
+		primeUsers := make([]*cardanofw.TestApexUser, instances)
+
+		maxWaitTime := 1200
+		successChance := 90 // 90%
+		succeededCount := int64(0)
+
+		fmt.Printf("Funding %v Wallets\n", instances)
+
+		for i := 0; i < instances; i++ {
+			primeUsers[i] = apex.CreateAndFundUser(t, ctx, uint64(10_000_000))
+			require.NotNil(t, primeUsers[i])
+		}
+
+		fmt.Printf("Funding complete\n")
+		fmt.Printf("Sending transactions\n")
+
+		var wg sync.WaitGroup
+		for i := 0; i < instances; i++ {
+			wg.Add(1)
+
+			go func(idx int) {
+				defer wg.Done()
+
+				if successChance > rand.Intn(100) {
+					succeededCount++
+					sleepTime := rand.Intn(maxWaitTime)
+					time.Sleep(time.Second * time.Duration(sleepTime))
+
+					txHash, err := primeUsers[idx].BridgeNexusAmount(t, ctx, txProviderPrime, apex.Bridge.PrimeMultisigAddr,
+						receiverAddrNexus, sendAmountDfm.Uint64(), apex.PrimeCluster.NetworkConfig(), receiverAddrNexus)
+					require.NoError(t, err)
+					fmt.Printf("Tx %v sent. hash: %s\n", idx+1, txHash)
+				} else {
+					const feeAmount = 1_100_000
+
+					receivers := map[string]uint64{
+						receiverAddrNexus: sendAmount * 10, // 10Ada
+					}
+
+					bridgingRequestMetadata, err := cardanofw.CreateMetaData(
+						primeUsers[idx].PrimeAddress, receivers, cardanofw.ChainIDNexus, feeAmount)
+					require.NoError(t, err)
+
+					txHash, err := cardanofw.SendTx(ctx, txProviderPrime, primeUsers[idx].PrimeWallet,
+						sendAmountDfm.Uint64()+feeAmount, apex.Bridge.PrimeMultisigAddr,
+						apex.PrimeCluster.NetworkConfig(), bridgingRequestMetadata)
+					require.NoError(t, err)
+
+					fmt.Printf("Tx %v sent without waiting for confirmation. hash: %s\n", idx+1, txHash)
+				}
+			}(i)
+		}
+
+		wg.Wait()
+
+		fmt.Printf("All tx sent, waiting for confirmation.\n")
+
+		var newAmount *big.Int
+
+		expectedAmount := new(big.Int).SetInt64(succeededCount)
+		expectedAmount.Mul(expectedAmount, sendAmountEth)
+		expectedAmount.Add(expectedAmount, ethBalanceBefore)
+
+		for i := 0; i < instances; i++ {
```

Comment:
it would be better if this can be executed in parallel for each instance. There are some useful functions for that in framework. Also thread sleep is not good because it does not count for context.Done()